### PR TITLE
Feature/16 kcl import

### LINKED ISSUE #16: KCL importer
The VKCO works well one first usage but lacks of import functionality against KCL (v1). So we need it for migration purposes.

### PR DESCRIPTION
**Pull request by wem**

### CODE REVIEW

File: src/main/kotlin/ch/sourcemotion/vertx/kinesis/consumer/orchestra/VertxKinesisOrchestraOptions.kt [hunk 1]
```diff
@@ -208,3 +217,19 @@ data class LoadConfiguration(val strategy: LoadStrategy, val exactCount: Int? =
...
+    val dynamoDbEndpoint: String? = null,
+
+    /**
+     * Make it possible to use DynamoDB specific aws credentials provider. If not defined,
+     * [VertxKinesisOrchestraOptions.credentialsProviderSupplier] will be used instead.
+     */
+    val credentialsProviderSupplier: Supplier<AwsCredentialsProvider>? = null
+) {
+    companion object {
+        const val DEFAULT_IMPORT_ADDRESS = "/kinesis-consumer-orchester/kcl-import"
```
Comment (wegenmic) [inline src/main/kotlin/ch/sourcemotion/vertx/kinesis/consumer/orchestra/VertxKinesisOrchestraOptions.kt]:
```suggestion
        const val DEFAULT_IMPORT_ADDRESS = "/kinesis-consumer-orchestra/kcl-import"
```

File: src/main/kotlin/ch/sourcemotion/vertx/kinesis/consumer/orchestra/consumer/AbstractKinesisConsumerVerticle.kt [hunk 2]
```diff
@@ -125,7 +131,7 @@ abstract class AbstractKinesisConsumerVerticle : CoroutineVerticle() {
         }
     }
 
-    private suspend fun startPolling(fetchStartPosition: FetchPosition) {
+    private suspend fun startFetching(fetchStartPosition: FetchPosition) {
         logger.debug { "Start polling for $consumerInfo" }
```
Comment (wegenmic) [inline src/main/kotlin/ch/sourcemotion/vertx/kinesis/consumer/orchestra/consumer/AbstractKinesisConsumerVerticle.kt]:
```suggestion
        logger.debug { "Start fetching for $consumerInfo" }
```

Comment (wegenmic) [reply src/main/kotlin/ch/sourcemotion/vertx/kinesis/consumer/orchestra/consumer/AbstractKinesisConsumerVerticle.kt]: There are a few other places in this class where the `poll` / `polling` naming is used. Should we adapt to stay consistent with the wording?

**Comment by wegenmic** [comment]
Here is an idea:

The `KCLV1Importer` is described as follows
> Importer of KCL checkpoint sequence values. This is useful to migrate from KCL v1 to VKCO.
> This way the event processing can be continued seamless after migration.

What if we have a built-in way to ensure backwards compatibility? This would allow a rollback to the KCL v1 way without losing the checkpoint information.

**Comment by wem** [comment]
> Here is an idea:
> 
> The `KCLV1Importer` is described as follows
> 
> > Importer of KCL checkpoint sequence values. This is useful to migrate from KCL v1 to VKCO.
> > This way the event processing can be continued seamless after migration.
> 
> What if we have a built-in way to ensure backwards compatibility? This would allow a rollback to the KCL v1 way without losing the checkpoint information.

Good question.

Basically, the readme/wiki must be finished asap after the first "final" state.

The user is able to provide its own shard persistence, so it's possible to implement a "merge persistence" this way. Another way could be to send each proceeded sequence number over the event bus, so the user can do some independent things with it.

**Pull request merged**